packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -22,7 +21,8 @@ import { OrderBook } from '../../api/Liquidity';
 import { ParsedLog } from '@augurproject/types';
 import { MarketReportingState, SECONDS_IN_A_DAY } from '../../constants';
 import { QUINTILLION, padHex } from '../../utils';
-import { Block } from "ethereumjs-blockstream";
+import { Block } from 'ethereumjs-blockstream';
+import { sleep } from '../utils/utils';
```

Comment:
code smell having to sleep

Reply:
Very true. There's a discussion about this in discord #dev. We'll need to rearchitect a good deal of our state management code to fix this without sleep and locking.